packages/synapse-react-client/src/components/TimelinePlot/TimelinePhase.tsx

```diff
@@ -235,6 +234,31 @@ const TimelinePhase = ({
     return getTimelineData(timepointData, rowData)
   }, [timepointData, rowData])
 
+  const observationIds = selectedRows
+    .map(row => row.values[0]) // Get observationId - identifier but not all rows have this
+    .filter(observationId => observationId)
+    .map(observationId => `'${observationId}'`)
+
+  const observationTexts = selectedRows
+    .filter(row => !row.values[0]) // Only take rows without observationId
+    .map(row => row.values[4]) // Get observationText - all rows have this
+    .filter(text => text)
+    .map(text => `'${text}'`)
+
+  let sql = ''
+
+  if (!observationIds.length) {
+    sql = `SELECT * FROM syn51735464
+    WHERE observationText IN (${observationTexts.join(', ')})`
+  } else {
+    sql = `SELECT * FROM syn51735464
+    WHERE observationId IN (${observationIds.join(', ')})${
+      observationTexts.length > 0
+        ? ` OR observationText IN (${observationTexts.join(', ')})`
+        : ''
+    }`
+  }
```

Comment:
Why do you need this SQL? Sorry, I don't totally understand the feature

Reply:
We can use the `ROW_ID` meta-column, so here we can collect the row IDs
```suggestion
  const selectedRowIds = selectedRows.map(row => row.rowId!)
```